linera-views/src/reentrant_collection_view.rs

```diff
@@ -506,17 +514,33 @@ where
                 btree_map::Entry::Occupied(entry) => {
                     let entry = entry.into_mut();
                     if let Update::Removed = entry {
-                        selected_short_keys.push((short_key, true));
+                        let key = self
+                            .context
+                            .base_tag_index(KeyTag::Subview as u8, &short_key);
+                        let context = self.context.clone_with_base_key(key);
+                        let view = W::new(context)?;
+                        let view = Arc::new(RwLock::new(view));
+                        *entry = Update::Set(view);
                     }
                 }
-                btree_map::Entry::Vacant(_entry) => {
-                    selected_short_keys.push((short_key, self.delete_storage_first));
+                btree_map::Entry::Vacant(entry) => {
+                    if self.delete_storage_first {
+                        let key = self
+                            .context
+                            .base_tag_index(KeyTag::Subview as u8, &short_key);
+                        let context = self.context.clone_with_base_key(key);
+                        let view = W::new(context)?;
+                        let view = Arc::new(RwLock::new(view));
+                        entry.insert(Update::Set(view));
+                    } else {
+                        selected_short_keys.push(short_key);
+                    }
                 }
             }
         }
         let mut handles = Vec::new();
```

Comment:
Could we create the handles directly instead of `selected_short_keys` then the handles?

Reply:
We could. But the next PR is to remove the handle from the code and replacing by a `pre_load` / `post_load`.

Reply:
Yes, we could. However, the next PR is about removing the handles altogether, so maybe not.